lang/java/avro/src/main/java/org/apache/avro/Conversions.java

```diff
@@ -77,6 +77,21 @@ public String getLogicalTypeName() {
       return "decimal";
     }
 
+    @Override
+    public BigDecimal fromCharSequence(CharSequence value, Schema schema, LogicalType type) {
+      BigDecimal result = new BigDecimal(value.toString());
+      int scale = ((LogicalTypes.Decimal) type).getScale();
```

Comment:
shouldn't there be validation of precision?

Reply:
thanks for bringing it up, that's a good idea. But right now we don't do it in other `from*` method. I thought we go to be consistent. 
I'd say we can create another PR/issue dedicated to the validation of precision and discuss it over there.